File: storage/data_transpiled/at_coder/abc121/D/4891128.java
public static int getA ( ) {
  int B = Integer . parseInt ( input . nextLine ( ) ) ;
  int count1 = Math . max ( 0 , ( B + 1 ) / 2 - ( A + 1 ) / 2 ) ;
  int res = count1 % 2 ;
  if ( A % 2 == 1 ) res ^= A ;
  if ( B % 2 == 0 ) res ^= B ;
  System . out . println ( res ) ;
  return 0 ;
}

File: storage/data_original/at_coder/abc121/D/4891128.py
A, B = map(int, input().split())

# (2n)^(2n+1) == 1.
# ??????????? XOR ??????0?????1
count1 = max(0, (B + 1) // 2 - (A + 1) // 2)
res = count1 % 2

# ???
if A % 2 == 1:
    res ^= A
if B % 2 == 0:
    res ^= B

print(res)